File: dashboard_backup_20260303/patch_ssg_cats.py
import os
import requests
from dotenv import load_dotenv

load_dotenv(os.path.join(os.path.dirname(os.path.abspath(__file__)), ".env"))

SUPABASE_URL = os.getenv("SUPABASE_URL")
SUPABASE_KEY = os.getenv("SUPABASE_KEY")

HEADERS = {
    "apikey": SUPABASE_KEY,
    "Authorization": f"Bearer {SUPABASE_KEY}",
    "Content-Type": "application/json",
    "Prefer": "return=representation"
}

NEW_CATEGORIES = [
    {"category_code": "BEAUTY", "name_ko": "뷰티", "sort_order": 101, "platform": "ssg"},
    {"category_code": "FASHION", "name_ko": "패션", "sort_order": 102, "platform": "ssg"},
    {"category_code": "LUXURY", "name_ko": "명품", "sort_order": 103, "platform": "ssg"},
    {"category_code": "KIDS", "name_ko": "유아동", "sort_order": 104, "platform": "ssg"},
    {"category_code": "SPORTS", "name_ko": "스포츠", "sort_order": 105, "platform": "ssg"},
    {"category_code": "FOOD_LIFE", "name_ko": "푸드&리빙", "sort_order": 106, "platform": "ssg"}
]

print("Patching SSG Categories into Supabase...")

for cat in NEW_CATEGORIES:
    res = requests.post(
        f"{SUPABASE_URL}/rest/v1/categories",
        headers=HEADERS,
        params={"on_conflict": "category_code"},
        json=cat
    )
    if res.status_code in (200, 201):
        print(f"✅ Upserted: {cat['category_code']} -> {cat['name_ko']}")
    else:
        print(f"❌ Error for {cat['category_code']}: {res.text}")

print("Category DB Patch Complete.")

File: dashboard_backup_20260303/patch_ssg_cats.js
const { createClient } = require('@supabase/supabase-js');
const dotenv = require('dotenv');
const fs = require('fs');
const path = require('path');

dotenv.config({ path: path.join(__dirname, '.env') });

const supabaseUrl = process.env.SUPABASE_URL;
const supabaseKey = process.env.SUPABASE_KEY;
const supabase = createClient(supabaseUrl, supabaseKey);

const newCategories = [
    { code: 'BEAUTY', name: '뷰티', display_order: 101 },
    { code: 'FASHION', name: '패션', display_order: 102 },
    { code: 'LUXURY', name: '명품', display_order: 103 },
    { code: 'KIDS', name: '유아동', display_order: 104 },
    { code: 'SPORTS', name: '스포츠', display_order: 105 },
    { code: 'FOOD_LIFE', name: '푸드&리빙', display_order: 106 }
];

async function patchCategories() {
    console.log("Patching SSG Categories into database...");

    for (const cat of newCategories) {
        const { data, error } = await supabase
            .from('categories')
            .upsert({
                code: cat.code,
                name: cat.name,
                display_order: cat.display_order,
                icon_name: 'category'
            }, { onConflict: 'code', ignoreDuplicates: false });

        if (error) {
            console.error(`Error upserting ${cat.code}:`, error.message);
        } else {
            console.log(`Successfully mapped ${cat.code} -> ${cat.name}`);
        }
    }
    console.log("Category DB Patch Complete.");
}

patchCategories();
